ARCHITECTURE.md

```diff
@@ -0,0 +1,197 @@
+The concrete semantics of code layout are completely at her choosing.
+For example, she could separate application tiers like frontend and backend into their own Cell, each.
+Or she could reflect the microservices architecture in the Cells.
+
+Paisano has a first class concept of Cells.
+By simply placing a folder in the repository, Paisano will pick it up.
+In that regard, Paisano is an automated importer, that spares the user the need to setup and maintain boilerplate plumbing code.
+
+Within a Cell, the user groups artifacts within Blocks of an appropriate Block Type.
+When configuring Standard, she names her Blocks using Standard's Block Types so that Paisano's importer can pick them up, too.
+By doing that, she also declares the repository's artifact type system to humans and machines.
+
+Machines can make great use of that to interact with the artifact type system in multiple ways.
+Paisano exports structured json-serializable data about a repository's _typed_ artifacts in its so-called "Paisano Registry".
+A CLI or TUI, as is bundled with Standard, or even a web user interface can consume, represent and act upon that data.
+
+And so can CI.
+
+In fact, this is an innovation in the SDLC space:
+We can devise an implementation of a CI which, by querying Paisano's Registry, autonomously discovers all work that needs to be done.
+Indeed, we made a reference implementation for GitHub Actions over at [`divnix/std-action`](https://github.com/divnix/paisano).
+To our knowledge, this is the first and only "zero config" CI implementation based on the principles of artifact typing and code organization.
+By using these principles rather than a rigid opinionated structure, it also remains highly flexible and adapts well to the user's preferences.
+
+In summary, all these organization and typing principles enable:
+
+- Easy refactoring of your repository's namespace for devops artifacts
+- Intuitive grouping of functionality that encourages well-defined internal boundaries
+  - Thereby keeping your automation code clean and maintainable
+- Making use of Block Types and the shared library to implement the DRY principle
+- Reasoning about the content of your repo through structured data
+  - Thereby interesting user interfaces, such as a CLI, TUI or even a UI
+  - But also services such as (close to) zero config & self-updating CI
+- Similar organizational principles help to lower the cost of context switching between different projects
+
+### Standard's Block Types (DevOps Type System)
+
+As mentioned above, Standard exploits the Block Type abstraction to provide artifact types for the SDLC.
+Within the semantics of each Block Type, we implement shared functionality.
+This is designed to offer the user an optimized, audited implementation.
+It also saves the need to devise "yet another" local implementation of otherwise well-understood generic functionality, such as, for example the building of a package or the pushing of a container image.
```

Comment:
"such as" and "for example" mean the same thing, so we only need one. either will do